src/extension/background-script/actions/accounts/add.ts

```diff
@@ -1,13 +1,33 @@
 import { v4 as uuidv4 } from "uuid";
 import { encryptData } from "~/common/lib/crypto";
 import state from "~/extension/background-script/state";
-import type { MessageAccountAdd } from "~/types";
+import type { Account, MessageAccountAdd } from "~/types";
 
 const add = async (message: MessageAccountAdd) => {
   const newAccount = message.args;
+  let name = newAccount.name;
   const accounts = state.getState().accounts;
   const tmpAccounts = { ...accounts };
 
+  // get all accounts names to avoid duplicate names
+  const accountNames = Object.values(accounts).map((el: Account) => el.name);
```

Comment:
Quite hard to understand what is happening here exactly. 🤔

Can we extract a method for this logic and add some unit tests? I'll try to wrap my head around it in the meantime 😅

Reply:
Also prettier seems to fail for this file, you might want to have a look at that.